src/utils/cluster_setup.py

```diff
@@ -0,0 +1,40 @@
+import click
+import subprocess
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.pass_context
+def cluster_setup(ctx: click.Context):
```

Comment:
Would that make sense to move this outside the `src` directory as it's not part of the provider sources and is meant for testing only?